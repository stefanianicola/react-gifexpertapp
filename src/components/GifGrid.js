import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);


    return (
        <>
            <h3>{category}</h3>

            {
                loading ? (<p>Cargando...</p>) : (
                    <div className='card-grid'>

                        <ol>
                            {
                                images.map(img => {
                                    return <GifGridItem
                                        key={img.id}
                                        {...img} />
                                })
                            }
                        </ol>
                    </div>
                )
            }

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
