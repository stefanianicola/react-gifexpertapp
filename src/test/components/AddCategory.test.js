import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('pruebas en AddCategory', () => {
    const setCategories = jest.fn();

    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    it('debe mostrar bien el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'holis';
        input.simulate('change', { target: { value } })
    });

    it('no debe postear informacion onSubmit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();
    });

    it('debe postear informacion onSubmit', () => {
        const input = wrapper.find('input');
        const value = 'holis';
        input.simulate('change', { target: { value } })
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).toHaveBeenCalled();
        expect(input.prop('value')).toBe('');
    });
})