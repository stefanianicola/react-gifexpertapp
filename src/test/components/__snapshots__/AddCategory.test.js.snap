// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en AddCategory debe mostrar bien el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
